k-distribution/include/kframework/builtin/domains.md

```diff
@@ -930,7 +934,7 @@ You can:
 ```k
   syntax Int ::= "~Int" Int                     [function, klabel(~Int_), symbol, functional, latex(\mathop{\sim_{\scriptstyle\it Int}}{#1}), hook(INT.not), smtlib(notInt)]
                > left:
-                 Int "^Int" Int                 [function, klabel(_^Int_), symbol, left, smt-hook(^), latex({#1}\mathrel{{\char`\^}_{\!\scriptstyle\it Int}}{#2}), hook(INT.pow)]
+                 Int "^Int" Int                 [function, klabel(_^Int_), symbol, left, smt-hook((ite (and (< 0 #1) (< 0 #2)) 0 (to_int (^ #1 #2)))), latex({#1}\mathrel{{\char`\^}_{\!\scriptstyle\it Int}}{#2}), hook(INT.pow)]
```

Comment:
I'm not sure what you're trying to encode here.  This encoding means `if (> #1 0) and (> #2 0), then 0, else (to_int (^ #1 #2))`, which doesn't sound correct to me.

Reply:
You might want to do this?
```suggestion
                 Int "^Int" Int                 [function, klabel(_^Int_), symbol, left, smt-hook((implies (and (< 0 #1) (< 0 #2)) (to_int (^ #1 #2)))), latex({#1}\mathrel{{\char`\^}_{\!\scriptstyle\it Int}}{#2}), hook(INT.pow)]
```

Reply:
Also, if you have `(< 0 #2)`, you don't need to add `(< 0 #1)`, right?  I mean, `X ^ N` will be well-defined for all X, if N > 0, right?